lib/action_policy/lookup_chain.rb

```diff
@@ -112,19 +112,19 @@ def policy_class_name_for(record)
     }
 
     # Infer from passed symbol
-    SYMBOL_LOOKUP = ->(record, namespace: nil, **) {
+    SYMBOL_LOOKUP = ->(record, namespace: nil, strict: false, **) {
       next unless record.is_a?(Symbol)
 
       policy_name = "#{record.camelize}Policy"
-      lookup_within_namespace(policy_name, namespace) || policy_name.safe_constantize
+      lookup_within_namespace(policy_name, namespace) || (strict ? nil : policy_name.safe_constantize)
```

Comment:
What about moving this option to `lookup_within_namespace`, so we can handle all the cases (currently, we do not respect this option for non-symbol/string lookups).

And we definitely need more specific name, say, `strict_namespace`; it should be clear that it only affects namespaces.

Reply:
@palkan done, let me know what you think.